ui/app/AppLayouts/Communities/views/CommunityColumnView.qml

```diff
@@ -510,22 +445,126 @@ Item {
+            anchors.topMargin: Style.current.halfPadding
+            anchors.bottomMargin: Style.current.halfPadding
+            anchors.horizontalCenter: parent.horizontalCenter
+            enabled: !root.communityData.amIBanned
+
+            text: {
+                if (root.communityData.amIBanned) return qsTr("You were banned from community")
+                if (invitationPending) return qsTr("Membership request pending...")
+
+                return root.communityData.access === Constants.communityChatOnRequestAccess ?
+                            qsTr("Request to join") : qsTr("Join Community")
+            }
+
+            onClicked: {
+                Global.openPopup(communityIntroDialog);
+            }
+
+            Connections {
+                enabled: joinCommunityButton.loading
+                target: root.store.communitiesModuleInst
+                function onCommunityAccessRequested(communityId: string) {
+                    if (communityId === communityData.id) {
+                        joinCommunityButton.invitationPending = root.store.isCommunityRequestPending(communityData.id)
+                        joinCommunityButton.loading = false
+                    }
+                }
+                function onCommunityAccessFailed(communityId: string) {
+                    if (communityId === communityData.id) {
+                        joinCommunityButton.invitationPending = false
+                        joinCommunityButton.loading = false
+                        Global.displayToastMessage(qsTr("Request to join failed"),
+                                                   qsTr("Please try again later"),
+                                                   "",
+                                                   false,
+                                                   Constants.ephemeralNotificationType.normal,
+                                                   "")
+                    }
+                }
+                function onUserAuthenticationCanceled() {
+                    joinCommunityButton.invitationPending = false
+                    joinCommunityButton.loading = false
+                }
+            }
+
+            Component {
+                id: communityIntroDialog
+                CommunityIntroDialog {
+
+                    isInvitationPending: joinCommunityButton.invitationPending
+                    requirementsCheckPending: root.store.requirementsCheckPending
+                    name: communityData.name
+                    introMessage: communityData.introMessage
+                    imageSrc: communityData.image
+                    accessType: communityData.access
+                    loginType: root.store.loginType
+                    walletAccountsModel: WalletStore.RootStore.nonWatchAccounts
+                    permissionsModel: {
+                        root.store.prepareTokenModelForCommunity(communityData.id)
+                        return root.store.permissionsModel
```

Comment:
I'm wondering why is this a 2 steps action. IMO the `prepare` step is an unnecessary detail for the visual components. Might be a good idea to replace it with a store function `getCommunityPermissionsModel(communityId)` and do the prepare inside (not necessarily in this PR). WDYT?

Reply:
Maybe because the operation itself is async?

Reply:
It's async and looks to me like a lazy load implementation. But I think it's better to isolate this inside the store.

Reply:
maybe hiding it in `getCommunityPermissionsModel` is a bit better but imo both options looks like a design problem. Ideally it should be enough to just access `permissionsModel` property without taking care about any implementation details in stores/backends.

Reply:
Good point 🤔! I don't have the details for that, just changed it's parent to be a component. I'll create a new issue to check it out in a separte task.